src/common-ssh/ssh.c

```diff
@@ -383,45 +395,23 @@ static int guac_common_ssh_authenticate(guac_common_ssh_session* common_session)
         /* Check if password auth is supported on the server */
         if (strstr(user_authlist, "password") != NULL) {
 
-            /* Attempt password authentication */
-            if (libssh2_userauth_password(session, user->username, user->password)) {
-
-                /* Abort on failure */
-                char* error_message;
-                libssh2_session_last_error(session, &error_message, NULL, 0);
-                guac_client_abort(client,
-                        GUAC_PROTOCOL_STATUS_CLIENT_UNAUTHORIZED,
-                        "Password authentication failed: %s", error_message);
-
-                return 1;
-            }
-
-            /* Password authentication succeeded */
-            return 0;
-
+            return password_auth_with_retry(common_session);
         }
 
         /* Check if keyboard-interactive auth is supported on the server */
         if (strstr(user_authlist, "keyboard-interactive") != NULL) {
 
             /* Attempt keyboard-interactive auth using provided password */
-            if (libssh2_userauth_keyboard_interactive(session, user->username,
-                        &guac_common_ssh_kbd_callback)) {
-
-                /* Abort on failure */
-                char* error_message;
-                libssh2_session_last_error(session, &error_message, NULL, 0);
-                guac_client_abort(client,
-                        GUAC_PROTOCOL_STATUS_CLIENT_UNAUTHORIZED,
-                        "Keyboard-interactive authentication failed: %s",
-                        error_message);
-
-                return 1;
+            int result = libssh2_userauth_keyboard_interactive(session, user->username,
+                        &guac_common_ssh_kbd_callback);
+
+            if (result == LIBSSH2_ERROR_AUTHENTICATION_FAILED) {
+                /* SPS changes to password authentication at this point */
```

Comment:
SPS -> server

Reply:
I guess I take it out from the commit message as well.